server/helpers/field/renderConditionalFields.ts

```diff
@@ -0,0 +1,41 @@
+import FormWizard from 'hmpo-form-wizard'
+
+export type FieldEntry = [string, FormWizard.Field]
+
+export default function renderConditionalFields(
+  req: FormWizard.Request,
+  [key, field]: FieldEntry,
+  allFieldsEntries: FieldEntry[],
+) {
+  if (!field.items) {
+    return [key, field]
+  }
+  const fields = Object.fromEntries(allFieldsEntries)
+
+  return [
+    key,
+    {
+      ...field,
+      items: field.items.map(item => {
+        const conditionalFields = [item.conditional || []].flat()
+        const components = conditionalFields.map(conditionalFieldKey => {
+          const conditionalField = field.prefix
+            ? fields[`${field.prefix}[${conditionalFieldKey}]`]
+            : fields[conditionalFieldKey as string]
+
+          if (!conditionalField) {
+            return undefined
+          }
+
+          return req.services.frontendComponentsService.getComponent(conditionalField.component, conditionalField)
```

Comment:
Ah the components for the fields are coming from Frontend Component Service 👌 

Does it have to be in `req.services`?
I do wonder if this could use the same mechanism we use the same APIs mechanism we use for stuff like the IncidentReporting/OffenderSearch API etc  🤔.

Reply:
I have mentioned this with Igor in a separate chat. I put it in a service as there are additional things being done as part of the functions and the api call defined in the client is used in there. Thought it may make sense to keep the client for dealing with the api calls and then having the services uses them and adds whatever additional functionality. But can make changes if preferred.

Reply:
but this line does not call the frontend components client or api. it calls the service method which renders a nunjucks template.

Reply:
this has been changed as discussed